contracts/SpokePool.sol

```diff
@@ -1221,37 +1421,75 @@ abstract contract SpokePool is
     }
 
-    function _verifyUpdateV3DepositMessage(
-        address depositor,
+    function _verifyUpdateV3DepositMessageHelper(
+        bytes32 depositor,
         uint32 depositId,
         uint256 originChainId,
         uint256 updatedOutputAmount,
-        address updatedRecipient,
+        bytes32 updatedRecipient,
         bytes memory updatedMessage,
+        bytes32 hashType,
         bytes memory depositorSignature
     ) internal view {
         // A depositor can request to modify an un-relayed deposit by signing a hash containing the updated
         // details and information uniquely identifying the deposit to relay. This information ensures
         // that this signature cannot be re-used for other deposits.
-        // Note: We use the EIP-712 (https://eips.ethereum.org/EIPS/eip-712) standard for hashing and signing typed data.
-        // Specifically, we use the version of the encoding known as "v4", as implemented by the JSON RPC method
-        // `eth_signedTypedDataV4` in MetaMask (https://docs.metamask.io/guide/signing-data.html).
         bytes32 expectedTypedDataV4Hash = _hashTypedDataV4(
-            // EIP-712 compliant hash struct: https://eips.ethereum.org/EIPS/eip-712#definition-of-hashstruct
             keccak256(
                 abi.encode(
-                    UPDATE_V3_DEPOSIT_DETAILS_HASH,
+                    hashType,
                     depositId,
                     originChainId,
                     updatedOutputAmount,
                     updatedRecipient,
                     keccak256(updatedMessage)
                 )
             ),
-            // By passing in the origin chain id, we enable the verification of the signature on a different chain
             originChainId
         );
-        _verifyDepositorSignature(depositor, expectedTypedDataV4Hash, depositorSignature);
+        _verifyDepositorSignature(depositor.toAddress(), expectedTypedDataV4Hash, depositorSignature);
+    }
+
+    function _verifyUpdateV3DepositMessage(
+        bytes32 depositor,
+        uint32 depositId,
+        uint256 originChainId,
+        uint256 updatedOutputAmount,
+        bytes32 updatedRecipient,
+        bytes memory updatedMessage,
+        bytes memory depositorSignature
+    ) internal view {
+        _verifyUpdateV3DepositMessageHelper(
+            depositor,
+            depositId,
+            originChainId,
+            updatedOutputAmount,
+            updatedRecipient,
+            updatedMessage,
+            UPDATE_V3_DEPOSIT_DETAILS_HASH,
+            depositorSignature
+        );
+    }
+
+    function _verifyUpdateV3DepositMessage(
+        address depositor,
+        uint32 depositId,
+        uint256 originChainId,
+        uint256 updatedOutputAmount,
+        address updatedRecipient,
+        bytes memory updatedMessage,
+        bytes memory depositorSignature
+    ) internal view {
+        _verifyUpdateV3DepositMessageHelper(
```

Comment:
do we really need this extra helper? might be simpler to change the interface of _verifyUpdateV3DepositMessage to accept additional hashType parameter (see comment above).

Reply:
`_verifyUpdateV3DepositMessage` is also overloaded with 2 implementations, one for address and one for `bytes32`. I added the helper to reduce code duplication. Without it each `_verifyUpdateV3DepositMessage` would need to implement `_hashTypedDataV4(keccak256(abi.encode...` does it make sense?

Reply:
I meant avoiding overload in _verifyUpdateV3DepositMessage and let its caller in the external function to do address->bytes32 conversion when needed, as well as pass the relevant hashType from constants.